### PR TITLE
Clean-up armeria-zookeeper

### PR DESCRIPTION
**Pull request by trustin**
- Revise Javadoc, comments and log messages
- Rename ZookeeperEndpointGroup to ZooKeeperEndpointGroup
- Rename ZkNodeValueConverter to ZooKeeperNodeValueConverter
- Rename some fields in ZooKeeperEndpointGroup
  - zkNode -> zkNodePath
  - zkHandlerProxy -> zkFuture
  - statesQueue -> stateQueue
- Reorder the methods in ZooKeeperEndpointGroup
- Improve the parsing of zkNode value
- Add ZkNodeConverter.convert() which accepts a String for convenience
- Add Endpoint.withWeight() to provide a way to create an endpoint with
  only hostname and weight
- Rewrite DefaultZkNodeValueConverterTest using AssertJ

/cc @jonefeewang

### CODE REVIEW

File: zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+ * A ZooKeeper-based {@link EndpointGroup} implementation. This {@link EndpointGroup} retrieves the list of
+ * {@link Endpoint}s from a ZooKeeper zNode and updates it when the value of the zNode changes. When a
+ * ZooKeeper session expires, it will automatically reconnect to the ZooKeeper with exponential retry delay,
+ * starting from 1 second up to 60 seconds.
+ */
+public class ZooKeeperEndpointGroup implements EndpointGroup {
+
+    private static final Logger logger = LoggerFactory.getLogger(ZooKeeperEndpointGroup.class);
+
+    private static final int MAX_RETRY_DELAY_MILLIS = 60 * 1000; //one minute
```
Comment (imasahiro) [inline zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java]: Put whitespace before `one`

File: zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    private int retryDelayMills = 1000; //start with one second
```
Comment (imasahiro) [inline zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java]: Put whitespace before `start`.

File: zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+            Code responseCode = get(responseCodeInt);
+            switch (responseCode) {
+                case OK:
+                    exists = true;
+                    break;
+                case NONODE:
+                    exists = false;
+                    break;
+                case SESSIONEXPIRED:
+                    // Ignore this and let the zNode Watcher process it first
```
Comment (imasahiro) [inline zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java]: Add period

File: zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+import com.linecorp.armeria.client.routing.EndpointGroup;
+import com.linecorp.armeria.client.routing.EndpointGroupException;
+
+/**
+ * A ZooKeeper-based {@link EndpointGroup} implementation. This {@link EndpointGroup} retrieves the list of
+ * {@link Endpoint}s from a ZooKeeper zNode and updates it when the value of the zNode changes. When a
+ * ZooKeeper session expires, it will automatically reconnect to the ZooKeeper with exponential retry delay,
+ * starting from 1 second up to 60 seconds.
+ */
+public class ZooKeeperEndpointGroup implements EndpointGroup {
```
Comment (imasahiro) [inline zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java]: Question: do we need to implements `SafeCloseable`?

Comment (trustin) [inline zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/ZooKeeperEndpointGroup.java]: `EndpointGroup` now extends `SafeCloseable`.

File: core/src/main/java/com/linecorp/armeria/client/Endpoint.java [hunk 5]
```diff
@@ -250,10 +268,13 @@ public String toString() {
     public boolean equals(Object obj) {
         if (this == obj) {
             return true;
-        } else if (obj instanceof Endpoint) {
+        }
+
+        if (obj instanceof Endpoint) {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/Endpoint.java]: Fixed

**Pull request merged**